replica/database.cc

```diff
@@ -2432,15 +2432,15 @@ static sstring get_snapshot_table_dir_prefix(const sstring& table_name) {
     return table_name + "-";
 }
 
-static sstring extract_cf_name(const sstring& directory_name) {
+static std::pair<sstring, utils::UUID> extract_cf_name_and_uuid(const sstring& directory_name) {
     // cf directory is of the form: 'cf_name-uuid'
-    // since cf_name may contain '-' characters, look for the last occurance of '-'
-    // in the directory entry name
-    auto pos = directory_name.find_last_of('-');
-    if (pos == sstring::npos) {
-        on_internal_error(dblog, format("table directory entry name '{}' is invalid: no '-' separator found", directory_name));
+    // uuid is assumed to be exactly 32 hex characters wide.
+    constexpr size_t uuid_size = 32;
+    ssize_t pos = directory_name.size() - uuid_size - 1;
+    if (pos <= 0 || directory_name[pos] != '-') {
+        on_internal_error(dblog, format("table directory entry name '{}' is invalid: no '-' separator found at pos {}", directory_name, pos));
```

Comment:
I want to verify something: Is there some API where this directory_name comes from the user input? If there is, it shouldn't be an on_internal_error but just a regular exception.

Reply:
No, the function is never called directly outside of the database layer,
and there it is called for "sanitized" directory entries that are listed in keyspace directories residing under the configured data directories.